--- 50-miniupnpd.orig	2009-11-03 13:38:49.000000000 +0300
+++ 50-miniupnpd	2009-11-03 13:31:38.000000000 +0300
@@ -1,5 +1,28 @@
 #!/bin/sh
 
+config_check_option() # option value param
+  {
+  local val
+  config_get val "$3" "$1"
+  test "$val" == "$2" && found="$found $3"
+  }
+
+config_check_list() # section param option value
+  {
+  local CONFIG_SECTIONS
+
+  config_load "$1"
+
+  local found='';
+  config_foreach "config_check_option \"$3\" \"$4\"" "$2"
+  test -n "$found" && echo "$found"
+  }
+
+upnpOn() # listName
+  {
+  test "$1" == "$2" || config_check_list multiwan upnpon iface "$1" > /dev/null
+  }
+
 /etc/init.d/miniupnpd enabled && {
 
 	local state="${ZONE}_${INTERFACE}"
@@ -8,14 +31,14 @@
 	config_load upnpd
 	config_get extif config external_iface
 
-	if [ "$ACTION" = "add" ] && [ "$INTERFACE" = "$extif" ]; then
+	if [ "$ACTION" = "add" ] && upnpOn "$INTERFACE" "$extif"; then
 
 		local active
 		config_get active "$state" ifname
 
 		[ -z "$active" ] && {
 			local ipaddr
-			config_get ipaddr "$extif" ipaddr
+			config_get ipaddr "$INTERFACE" ipaddr
 
 			logger -t "upnp" "adding $INTERFACE ($DEVICE - $ipaddr) to firewall"
 
@@ -29,7 +52,7 @@
 			uci_set_state upnpd "$state" ipaddr "$ipaddr"
 		}
 	
-	elif [ "$ACTION" = "remove" ] && [ "$INTERFACE" = "$extif" ]; then
+	elif [ "$ACTION" = "remove" ] && upnpOn "$INTERFACE" "$extif"; then
 
 		local ifname ipaddr
 		config_get ifname "$state" ifname
